protocol/messenger_handler.go

```diff
@@ -1376,6 +1376,15 @@ func (m *Messenger) HandleCommunityCancelRequestToJoin(state *ReceivedMessageSta
 
 // HandleCommunityRequestToJoin handles an community request to join
 func (m *Messenger) HandleCommunityRequestToJoin(state *ReceivedMessageState, signer *ecdsa.PublicKey, requestToJoinProto protobuf.CommunityRequestToJoin) error {
+	if state.Response.RequestsToJoinCommunity != nil {
+		for _, request := range state.Response.RequestsToJoinCommunity {
+			if request.PublicKey == common.PubkeyToHex(signer) {
+				// We have already handled this request to join
```

Comment:
Users send the request to all admins and owners + the control node, so the owner might receive the same request twice

Reply:
if it will be received twice - it will be just replaced, correct? (after fix in DB querry)
Also maybe we can filter it out by the receiver. As I understand, in the case of the control node it will be a community

Reply:
Yeah, the DB only saved once. I did this so we don't add the request to join twice in the Response (signal to the client), because the client (desktop) is a bit dumb right now and just adds the request blindly. 

I guess you're right, we should just add some logic in the client to not have duplicate requests.